tests/base.py

```diff
@@ -182,6 +194,16 @@ def parse_date(date_value):
 
         raise NotImplementedError("Tests do not account for dates of this format: {}".format(date_value))
 
+    def convert_state_to_utc(self, date_str):
+        """
+        Convert a saved bookmark value of the form '2020-08-25T13:17:36-07:00' to
+        a string formatted utc datetime,
+        in order to compare aginast json formatted datetime values
+        """
+        date_object = dateutil.parser.parse(date_str)
+        date_object_utc = date_object.astimezone(tz=pytz.UTC)
+        return dt.strftime(date_object_utc, "%Y-%m-%dT%H:%M:%SZ")
+
 
     def timedelta_formatted(self, dtime, days=0):
```

Comment:
Update this function to provide the desired format as well in arguments. This will make sure that start_date, replication_key, and bookmark all are in a valid format. The above function will get passed if a bookmark is in start_date format too.

Reply:
Replaced above hardcoded value with valid bookmark format.

Note, we can't generalise both start date and bookmarks since we use below 2 different date formats for them,

```
    START_DATE_FORMAT = "%Y-%m-%dT00:00:00Z"
    BOOKMARK_DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
```